src/routes/files/handlers.js

```diff
@@ -0,0 +1,109 @@
+import { v4 as uuidv4 } from 'uuid';
+import * as fs from 'fs';
+import handleErrors from '../../lib/apiErrorHandler';
+import { sequelize, File } from '../../models';
+import s3Uploader from '../../lib/s3Uploader';
+
+const fileType = require('file-type');
+const multiparty = require('multiparty');
+
+const namespace = 'SERVICE:FILES';
+
+const ATTACHMENT = 'ATTACHMENT';
+const RESOURCE = 'RESOURCE';
+
+const logContext = {
+  namespace,
+};
+
+export const createFileMetaData = async (
+  originalFileName,
+  s3FileName,
+  reportId,
+  attachmentType,
+) => {
+  const newFile = {
+    activityReportId: reportId,
+    originalFileName,
+    attachmentType,
+    key: s3FileName,
+    status: 'UPLOADING',
+  };
+  let file;
+  try {
+    await sequelize.transaction(async (transaction) => {
+      file = await File.create(newFile, transaction);
+    });
+    return file.dataValues;
+  } catch (error) {
+    return error;
+  }
+};
+
+export const updateStatus = async (fileId, fileStatus) => {
+  let file;
+  try {
+    await sequelize.transaction(async (transaction) => {
+      file = await File.update({ status: fileStatus }, { where: { id: fileId } }, transaction);
+    });
+    return file.dataValues;
+  } catch (error) {
+    return error;
+  }
+};
+
+export default async function uploadHandler(req, res) {
+  const form = new multiparty.Form();
+  form.parse(req, async (error, fields, files) => {
+    if (error) {
+      res.status(500).send(error);
+    }
+    let buffer;
+    let metadata;
+    let fileName;
+    let type;
+
+    try {
+      if (!files.File) {
+        res.status(400).send({ error: 'file required' });
+        return;
+      }
+      const { path, originalFilename } = files.File[0];
+      const { reportId, attachmentType } = fields;
+      if (!reportId) {
+        res.status(400).send({ error: 'reportId required' });
+        return;
+      }
+      if (!attachmentType) {
+        res.status(400).send({ error: 'attachmentType required' });
+        return;
+      }
+      if (attachmentType[0] !== ATTACHMENT && attachmentType[0] !== RESOURCE) {
+        res.status(400).send({ error: `incorrect attachmentType. Wanted: ${ATTACHMENT} or ${RESOURCE}. Got: ${attachmentType[0]}` });
+        return;
+      }
+      buffer = fs.readFileSync(path);
+      type = await fileType.fromFile(path);
+      if (!type) {
+        res.status(400).send('Could not determine file type');
+        return;
+      }
```

Comment:
file-type doesn't support plain text formats. We need to decide on allowed file formats and have some sort of validation in the future. I think that belongs in a separate story though.